junos/func_common.go

```diff
@@ -90,6 +92,30 @@ func validateCIDRNetwork(network string) error {
 	return nil
 }
 
+func validateByteString(length int) schema.SchemaValidateDiagFunc {
+	return func(i interface{}, path cty.Path) diag.Diagnostics {
+		var diags diag.Diagnostics
+		regLength := length - 1
+		l := strconv.Itoa(regLength)
+		s := i.(string)
+		// regString := "^(\\d+:){" + l + "," + l + "}\\d+$"
+		regString := "^([\\d\\w]{2}:){" + l + "}[\\d\\w]{2}$"
+		r, err := regexp.Compile(regString)
+		regMatch := r.MatchString(s)
+
+		if err != nil || !regMatch {
+			//	return fmt.Errorf("%v is not a valid byte string", byteString)
+			diags = append(diags, diag.Diagnostic{
+				Severity:      diag.Error,
+				Summary:       fmt.Sprintf("%s is not a valid byte string, should be %s bytes long", i, strconv.Itoa(length)),
+				AttributePath: path,
+			})
+		}
+
+		return diags
+	}
+}
+
```

Comment:
Can you use builtin sdk function https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2@v2.4.0/helper/validation#StringMatch instead ?